AnimationGroup.lua

```diff
@@ -60,28 +60,55 @@ function CreateFrame(...)
     return frame
 end
 
-
 local function OnUpdate(self, elapsed)
     if self.paused then
         return
     end
-
-    self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+    local reverse = self.group.reverse
+    local in_start_delay = self.startDelay and self.startDelayTime < self.startDelay
+    local in_end_delay = self.endDelay and self.endDelayTime < self.endDelay
```

Comment:
These would be

```suggestion
    local is_start_delayed = self.startDelay and self.startDelayTime < self.startDelay
    local is_end_delayed = self.endDelay and self.endDelayTime < self.endDelay
```

am I right? In which case you can simplify `self.delaying` (which should probably be `self.delayed`):

```
self.delayed = is_start_delayed or is_end_delayed
```

Reply:
no its not that simple. its depends on reverse and animation completion